aten/src/ATen/native/cuda/MultiTensorApply.cuh

```diff
@@ -46,13 +56,72 @@ multi_tensor_apply_kernel(
   callable(kChunkSize, tensorListMeta, args...); 
 }
 
+template<int depth, typename T, typename... ArgTypes>
+void multi_tensor_apply(
+    std::vector<std::vector<at::Tensor>>& tensor_lists,
+    at::ArrayRef<double> scalars,
+    T callable,
+    ArgTypes... args) {
+        TORCH_CHECK(tensor_lists.size() == depth, "Number of tensor lists has to match the depth.");
+
+        size_t n_tensors = tensor_lists[0].size();
+        TensorListScalarListMetadata<depth> tensorListMeta;
+
+        int loc_block_info = 0;
+        int loc_tensor_info = 0;
+        for(size_t t = 0; t < n_tensors; t++) {
+
+            tensorListMeta.scalar_vals[t] = scalars[t];
```

Comment:
Use
```
tensorListMeta.scalar_vals[loc_tensor_info] = scalars[t];
```
just as we use `loc_tensor_info` when populating sizes` and `addresses`.  `loc_tensor_info` tracks where in the metadata struct we're putting this tensor's info.  For MTA launches that require several kernels, `loc_tensor_info` gets reset to zero several times.